src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,31 @@
+package pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class BasePage {
+    WebDriver driver;
+    WebDriverWait wait;
+    Actions actions;
+
+public BasePage(WebDriver givenDriver){
+    driver = givenDriver;
+    wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+    actions = new Actions(driver);
+    PageFactory.initElements(driver, this);
+}
+public void doubleClick(WebElement locator){
+wait.until(ExpectedConditions.elementToBeClickable(locator));
+actions.doubleClick().perform();
+}
+}
+
+
+
+
```

Comment:
**Please delete the empty line!** An Empty Line is a "Code Smell"
The short answer is, _They’re more an annoyance than a problem.They won’t actually break anything—which is why your code continues to run—but they make it awkward to read and change your code, especially if you’re dealing with multiple line breaks per line of real code. They also add just a tiny bit of size to your files, which is nice to reduce._